File: app/src/main/java/com/example/animabuilder/Activities/Fragments/SubFragments/SecondaryTable.java
package com.example.animabuilder.Activities.Fragments.SubFragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.TableLayout;
import android.widget.TableRow;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.example.animabuilder.CharacterCreation.Attributes.SecondaryCharacteristic;
import com.example.animabuilder.CharacterCreation.SecondaryInput;
import com.example.animabuilder.R;

import com.example.animabuilder.CharacterCreation.BaseCharacter;

public class SecondaryTable extends Fragment {
    BaseCharacter charInstance;
    int rID;

    public SecondaryTable(BaseCharacter charInstance, int rID){
        this.charInstance = charInstance;
        this.rID = rID;
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState){
        super.onCreate(savedInstanceState);
        View view = inflater.inflate(rID, container, false);

        TableLayout workingTable= view.findViewById(R.id.heldTable);

        for(int i = 1; i < workingTable.getChildCount(); i++){
            TableRow workingRow = (TableRow) workingTable.getChildAt(i);

            SecondaryCharacteristic workingStat;
            String rowText = ((TextView)workingRow.getChildAt(0)).getText().toString();

            switch(rowText){
                case "Acrobatics": workingStat = charInstance.getSecondaryList().getAcrobatics(); break;
                case "Athletics": workingStat = charInstance.getSecondaryList().getAthletics(); break;
                case "Climb": workingStat = charInstance.getSecondaryList().getClimb(); break;
                case "Jump": workingStat = charInstance.getSecondaryList().getJump(); break;
                case "Ride": workingStat = charInstance.getSecondaryList().getRide(); break;
                case "Swim": workingStat = charInstance.getSecondaryList().getSwim(); break;

                case "Intimidate": workingStat = charInstance.getSecondaryList().getIntimidate(); break;
                case "Leadership": workingStat = charInstance.getSecondaryList().getLeadership(); break;
                case "Persuasion": workingStat = charInstance.getSecondaryList().getPersuasion(); break;
                case "Style": workingStat = charInstance.getSecondaryList().getStyle(); break;

                case "Notice": workingStat = charInstance.getSecondaryList().getNotice(); break;
                case "Search": workingStat = charInstance.getSecondaryList().getSearch(); break;
                case "Track": workingStat = charInstance.getSecondaryList().getTrack(); break;

                case "Animals": workingStat = charInstance.getSecondaryList().getAnimals(); break;
                case "Appraise": workingStat = charInstance.getSecondaryList().getAppraise(); break;
                case "Herb Lore": workingStat = charInstance.getSecondaryList().getHerbalLore(); break;
                case "History": workingStat = charInstance.getSecondaryList().getHistory(); break;
                case "Memorize": workingStat = charInstance.getSecondaryList().getMemorize(); break;
                case "Magic Appraise": workingStat = charInstance.getSecondaryList().getMagicAppraise(); break;
                case "Medicine": workingStat = charInstance.getSecondaryList().getMedic(); break;
                case "Navigation": workingStat = charInstance.getSecondaryList().getNavigate(); break;
                case "Occult": workingStat = charInstance.getSecondaryList().getOccult(); break;
                case "Science": workingStat = charInstance.getSecondaryList().getSciences(); break;

                case "Composure": workingStat = charInstance.getSecondaryList().getComposure(); break;
                case "Feats of Strength": workingStat = charInstance.getSecondaryList().getStrengthFeat(); break;
                case "Withstand Pain": workingStat = charInstance.getSecondaryList().getResistPain(); break;

                case "Disguise": workingStat = charInstance.getSecondaryList().getDisguise(); break;
                case "Hide": workingStat = charInstance.getSecondaryList().getHide(); break;
                case "Lock Picking": workingStat = charInstance.getSecondaryList().getLockPick(); break;
                case "Poisons": workingStat = charInstance.getSecondaryList().getPoisons(); break;
                case "Theft": workingStat = charInstance.getSecondaryList().getTheft(); break;
                case "Stealth": workingStat = charInstance.getSecondaryList().getStealth(); break;
                case "Trap Lore": workingStat = charInstance.getSecondaryList().getTrapLore(); break;

                case "Art": workingStat = charInstance.getSecondaryList().getArt(); break;
                case "Dance": workingStat = charInstance.getSecondaryList().getDance(); break;
                case "Forging": workingStat = charInstance.getSecondaryList().getForging(); break;
                case "Music": workingStat = charInstance.getSecondaryList().getMusic(); break;
                case "Sleight of Hand": workingStat = charInstance.getSecondaryList().getSleightHand(); break;

                default: workingStat = new SecondaryCharacteristic(); break;
            }

            ((EditText)workingRow.getChildAt(1)).setText(getString(R.string.intItem, workingStat.getPointsApplied()));
            ((EditText)workingRow.getChildAt(1)).addTextChangedListener(new SecondaryInput(workingRow, view.getContext(), workingStat));

            ((TextView)workingRow.getChildAt(2)).setText(getString(R.string.intItem, workingStat.getModVal()));
            ((TextView)workingRow.getChildAt(3)).setText(getString(R.string.intItem, workingStat.getPointsFromClass()));
            if(workingStat.isBonusApplied())
                ((TextView)workingRow.getChildAt(4)).setText("5");
            else
                ((TextView)workingRow.getChildAt(4)).setText("0");
            ((TextView)workingRow.getChildAt(5)).setText(getString(R.string.intItem, workingStat.getTotal()));
        }

        return view;
    }
}

File: app/src/main/java/com/example/animabuilder/Activities/Fragments/SubFragments/SecondaryTable.kt
package com.example.animabuilder.Activities.Fragments.SubFragments

import com.example.animabuilder.CharacterCreation.BaseCharacter
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.view.View
import android.widget.TableLayout
import com.example.animabuilder.R
import com.example.animabuilder.CharacterCreation.Attributes.SecondaryCharacteristic
import android.widget.TextView
import android.widget.EditText
import android.widget.TableRow
import androidx.fragment.app.Fragment
import com.example.animabuilder.SecondaryInput

class SecondaryTable(var charInstance: BaseCharacter, var rID: Int) : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {

        super.onCreate(savedInstanceState)
        val view = inflater.inflate(rID, container, false)
        val workingTable = view.findViewById<TableLayout>(R.id.heldTable)

        for (i in 1 until workingTable.childCount) {
            val workingRow = workingTable.getChildAt(i) as TableRow
            var workingStat: SecondaryCharacteristic
            val rowText = (workingRow.getChildAt(0) as TextView).text.toString()

            workingStat = when (rowText) {
                getString(R.string.acrobaticsLabel) -> charInstance.secondaryList.acrobatics
                getString(R.string.athleticsLabel) -> charInstance.secondaryList.athletics
                getString(R.string.climbLabel) -> charInstance.secondaryList.climb
                getString(R.string.jumpLabel) -> charInstance.secondaryList.jump
                getString(R.string.rideLabel) -> charInstance.secondaryList.ride
                getString(R.string.swimLabel) -> charInstance.secondaryList.swim

                getString(R.string.intimidateLabel) -> charInstance.secondaryList.intimidate
                getString(R.string.leadershipLabel) -> charInstance.secondaryList.leadership
                getString(R.string.persuasionLabel) -> charInstance.secondaryList.persuasion
                getString(R.string.styleLabel) -> charInstance.secondaryList.style

                getString(R.string.noticeLabel) -> charInstance.secondaryList.notice
                getString(R.string.searchLabel) -> charInstance.secondaryList.search
                getString(R.string.trackLabel) -> charInstance.secondaryList.track

                getString(R.string.animalLabel) -> charInstance.secondaryList.animals
                getString(R.string.appraiseLabel) -> charInstance.secondaryList.appraise
                getString(R.string.herbalLabel) -> charInstance.secondaryList.herbalLore
                getString(R.string.histLabel) -> charInstance.secondaryList.history
                getString(R.string.memLabel) -> charInstance.secondaryList.memorize
                getString(R.string.mAppraiseLabel) -> charInstance.secondaryList.magicAppraise
                getString(R.string.medLabel) -> charInstance.secondaryList.medic
                getString(R.string.navLabel) -> charInstance.secondaryList.navigate
                getString(R.string.occultLabel) -> charInstance.secondaryList.occult
                getString(R.string.scienceLabel) -> charInstance.secondaryList.sciences

                getString(R.string.composureLabel) -> charInstance.secondaryList.composure
                getString(R.string.strFeatLabel) -> charInstance.secondaryList.strengthFeat
                getString(R.string.resistPainLabel) -> charInstance.secondaryList.resistPain

                getString(R.string.disguiseLabel) -> charInstance.secondaryList.disguise
                getString(R.string.hideLabel) -> charInstance.secondaryList.hide
                getString(R.string.lockpickLabel) -> charInstance.secondaryList.lockPick
                getString(R.string.poisonLabel) -> charInstance.secondaryList.poisons
                getString(R.string.theftLabel) -> charInstance.secondaryList.theft
                getString(R.string.stealthLabel) -> charInstance.secondaryList.stealth
                getString(R.string.trapLabel) -> charInstance.secondaryList.trapLore

                getString(R.string.artLabel) -> charInstance.secondaryList.art
                getString(R.string.danceLabel) -> charInstance.secondaryList.dance
                getString(R.string.forgeLabel) -> charInstance.secondaryList.forging
                getString(R.string.musicLabel) -> charInstance.secondaryList.music
                getString(R.string.sleightLabel) -> charInstance.secondaryList.sleightHand

                else -> SecondaryCharacteristic()
            }
            (workingRow.getChildAt(1) as EditText).setText(
                getString(
                    R.string.intItem,
                    workingStat.pointsApplied
                )
            )
            (workingRow.getChildAt(1) as EditText).addTextChangedListener(
                SecondaryInput(
                    workingRow,
                    view.context,
                    workingStat,
                    charInstance
                )
            )
            (workingRow.getChildAt(2) as TextView).text =
                getString(R.string.intItem, workingStat.modVal)
            (workingRow.getChildAt(3) as TextView).text =
                getString(R.string.intItem, workingStat.pointsFromClass)
            if (workingStat.isBonusApplied) (workingRow.getChildAt(4) as TextView).text =
                "5" else (workingRow.getChildAt(4) as TextView).text = "0"
            (workingRow.getChildAt(5) as TextView).text =
                getString(R.string.intItem, workingStat.total)
        }
        return view
    }
}